assets/js/base/context/cart-checkout/payment-methods/payment-method-data-context.js

```diff
@@ -103,26 +104,24 @@ const isErrorResponse = ( response ) => {
  * @param {Object} props                     Incoming props for provider
  * @param {Object} props.children            The wrapped components in this
  *                                           provider.
- * @param {string} props.activePaymentMethod The initial active payment method
- *                                           to set for the context.
  */
-export const PaymentMethodDataProvider = ( {
-	children,
-	activePaymentMethod: initialActivePaymentMethod,
-} ) => {
+export const PaymentMethodDataProvider = ( { children } ) => {
 	const { setBillingData } = useBillingDataContext();
 	const {
 		isComplete: checkoutIsComplete,
 		isProcessingComplete: checkoutIsProcessingComplete,
 		isCalculating: checkoutIsCalculating,
 		hasError: checkoutHasError,
 	} = useCheckoutContext();
-	const [ activePaymentMethod, setActive ] = useState(
-		initialActivePaymentMethod
-	);
+	const [ activePaymentMethod, setActive ] = useState( {} );
 	const [ observers, subscriber ] = useReducer( emitReducer, {} );
 	const currentObservers = useRef( observers );
-	const customerPaymentMethods = getSetting( 'customerPaymentMethods', {} );
+
+	const { isEditor, previewData } = useEditorContext();
+	const customerPaymentMethods =
+		isEditor && previewData?.previewSavedPaymentMethods
+			? previewData?.previewSavedPaymentMethods
+			: getSetting( 'customerPaymentMethods', {} );
```

Comment:
@senadir your suggestion won't work because `previewData` is an object itself exposed on `useEditorContext`.  I don't want to change the structure to a flat object because I think editor context should be block agnostic (and thus `previewData` can contain any preview data for any block. Flattening the object means we have an arbitrary set of props exposed by the context instead of just _one_ prop (`previewData`) that contains that arbitrary data.  So, what @haszari has here is fine.